framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4240,6 +4291,202 @@ private void makeGroundTargetType(
+            if (TypesUtils.isCaptured(capturedTypeArg.getUnderlyingType())
+                    && preCapturedArg.getKind() == TypeKind.WILDCARD) {
+                // The type argument is a captured type. Use the type argument from the newly
+                // created and yet-to-be annotated captureType. (The annotations are added as part
+                // of capturing the wildcard.)
+                typeVarToTypeArguments.put(typeVar, capturedTypeArg);
+                // Also, add a mapping from the captured type mirror to the annotated captured
+                // type, so that if one captured type refers to another, the correct annotated type
+                // is used.
+                captureToAnnotatedCapture.put(
+                        ((AnnotatedTypeVariable) capturedTypeArg).getUnderlyingType(),
+                        capturedTypeArg);
+            } else {
+                // The type argument is not a captured type. Use the type argument from
+                // typeToCapture, which is fully-annotated.
+                typeVarToTypeArguments.put(typeVar, preCapturedArg);
+            }
+        }
+
+        // Use the mapping above to substitute the type variables in capturedType and
+        // typeDeclaration.
+        capturedType =
+                (AnnotatedDeclaredType)
+                        typeVarSubstitutor.substituteWithoutCopyingTypeArguments(
+                                typeVarToTypeArguments, capturedType);
+
+        // Loop through the type arguments and set the annotations of each captured type.
+        List<AnnotatedTypeMirror> newTypeArgs = new ArrayList<>(typeToCapture.typeArgs);
+        List<TypeVariable> orderToCapture =
+                TypesUtils.order(captureToAnnotatedCapture.keySet(), types);
+        for (TypeVariable capture : orderToCapture) {
+            AnnotatedTypeMirror typeArg = captureToAnnotatedCapture.get(capture);
+            int i = capturedTypeMirror.getTypeArguments().indexOf(capture);
+            AnnotatedTypeMirror annoTypeArg = typeToCapture.getTypeArguments().get(i);
+            AnnotatedTypeVariable typeVariable =
+                    (AnnotatedTypeVariable) typeDeclaration.getTypeArguments().get(i);
+            captureWildcard(
+                    typeVarToTypeArguments,
+                    captureToAnnotatedCapture,
+                    (AnnotatedWildcardType) annoTypeArg,
+                    typeVariable,
+                    (AnnotatedTypeVariable) typeArg);
+            newTypeArgs.remove(i);
+            newTypeArgs.add(i, typeArg);
+        }
+        capturedType.setTypeArguments(newTypeArgs);
+
+        capturedType.addAnnotations(typeToCapture.getAnnotations());
+        return capturedType;
+    }
+
+    /**
+     * Set the annotated bounds for fresh type variable {@code capturedArg}, so that it is the
+     * capture of {@code wildcard}.
+     *
+     * @param argMapping substitution
```

Comment:
This Javadoc comment isn't descriptive, especially since the next formal parameter has the same type.